src/vmm/src/guest_config/templates/mod.rs

```diff
@@ -0,0 +1,193 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::str::FromStr;
+
+use serde::de::Error;
+use serde::{Deserialize, Deserializer};
+
+/// Types common for template definitions for both
+/// Aarch64 and x86_64.
+
+/// Bit-mask based wrapper to apply
+/// changes to a given register's value
+#[derive(Debug, Deserialize, Eq, PartialEq)]
+pub struct RegisterModifier {
+    /// Pointer of the location to be bit masked.
+    #[serde(deserialize_with = "deserialize_u64_from_str")]
+    addr: u64,
+    /// Mask to be applied to register's value at the address
+    /// provided.
+    mask: BitMask,
+}
+
+/// Bit mask to adjust targeted bits to the value,
+/// applied by the filter only.
+#[derive(Debug, Deserialize, Eq, PartialEq)]
+pub struct BitMask {
```

Comment:
I don't really like `BitMask` encapsulating `filter` and `value`. Can we place `filter` (or `mask`) and `value` directly to the `RegisterModifier`?

Reply:
It might be difficult to do that, because it needs to associate a custom deserialization function with a field...
```
    /// Mask to be applied to register's value at the address
    /// provided.
    #[serde(deserialize_with = "deserialize_bitmask")]
    mask: BitMask,
```

Reply:
I see, makes sense, `filter` and `value` need to go together in a struct.
The problem with naming arises because `RegisterModifier` is good for both (addr, filter, value) and (filter, value).
If `RegisterModifier` is used for (addr, filter, value), it's hard to come up with a sensible name for (filter, value).

Reply:
What do you think about the following naming?
```
struct CpuidModifier {
    mask: u32,
    value: u32,
}

#[derive(Deserialize, Debug)]
struct CpuidTemplate {
    leaf: u32,
    subleaf: u32,
    #[serde(deserialize_with = "deserialize_cpuid_register")]
    register: CpuidRegister,
    #[serde(deserialize_with = "deserialize_u32_modifier")]
    modifier: CpuidModifier,
}

#[derive(Deserialize, Debug)]
struct X86_64CpuTemplate {
    cpuid_template: Vec<CpuidTemplate>,
    msr_template: Vec<MsrTemplate>,
}
```